dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyServerHandler.java

```diff
@@ -102,6 +106,20 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
         }
     }
 
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+        NettyChannel channel = NettyChannel.getOrAddChannel(ctx.channel(), url, handler);
+        try {
+            if (evt instanceof IdleStateEvent) {
```

Comment:
Seems like we should keep the logic same with NettyClientHandler. We can call `NettyChannel.getOrAddChannel` after `if (evt instanceof IdleStateEvent)`

Reply:
Can simply use `int heartbeatInterval = UrlUtils.getHeartbeat(getUrl()) * 3;` to fix this problem? @chickenlj

Reply:
Yes, I notice @beiwei30 has just fixed it using getIdleTimeout.